### PR TITLE
Extracted the json string from the script_output output

### PR DESCRIPTION
**Pull request by tinawang123**
Extracted the json string from the script_output output. In case the output include unexpected information
It is related with SUSEConnect -s before and after migration check. Use script_output may get some dhcp information. But I only need the json string, so need parse the output information and get the exact return value.

- Related ticket: https://progress.opensuse.org/issues/58241
- Verification run: https://openqa.suse.de/tests/3487181#step/install_service/479
Please check the attachment
[serial0.txt](https://github.com/os-autoinst/os-autoinst-distri-opensuse/files/3742415/serial0.txt)

### CODE REVIEW

File: lib/services/registered_addons.pm [hunk 1]
```diff
@@ -64,8 +64,9 @@ sub check_upgraded_addons {
 }
 
 sub check_suseconnect {
-    my $output = script_output("SUSEConnect -s");
-    my $json   = Mojo::JSON::decode_json($output);
+    my $output = script_output("SUSEConnect -s", 120);
+    my @out    = grep { $_ =~ /identifier/ } split(/\n/, $output);
+    my $json   = Mojo::JSON::decode_json($out[0]);
```
Comment (Amrysliu) [inline lib/services/registered_addons.pm]: Are you sure if the output will always include `identifier`? if there is no `identifier` string in the output, the `@out` will be an empty array.  I do not know how the `decode_json` to deal with an `undef`.

Comment (tinawang123) [reply lib/services/registered_addons.pm]: Add the condition to confirm the returned value

File: lib/services/registered_addons.pm [hunk 2]
```diff
@@ -64,20 +64,26 @@ sub check_upgraded_addons {
 }
 
 sub check_suseconnect {
-    my $output = script_output("SUSEConnect -s");
-    my $json   = Mojo::JSON::decode_json($output);
-    foreach (@$json) {
-        my $iden   = $_->{identifier};
-        my $status = $_->{status};
-        die "$iden register status is: $status" if ($status ne 'Registered');
+    my $output = script_output("SUSEConnect -s", 120);
```
Comment (okurz) [inline lib/services/registered_addons.pm]: why not just `assert_script_run 'SUSEConnect --status-text | (! grep -q "Not Registered")';` instead of the whole block?

File: lib/services/registered_addons.pm [hunk 3]
```diff
@@ -64,20 +64,26 @@ sub check_upgraded_addons {
...
+        die "Cannot get register status: $output";
     }
 }
 
 sub full_registered_check {
     my ($stage) = @_;
     $stage //= '';
-    check_suseconnect();
     zypper_lr();
+    check_suseconnect();
```
Comment (okurz) [inline lib/services/registered_addons.pm]: this does not look related

Comment (tinawang123) [reply lib/services/registered_addons.pm]: We want to check register addons and register status. As there is one bug: https://bugzilla.suse.com/show_bug.cgi?id=1149719
Which found by customer at first. So, we want to check 'SUSEConnect -s ' before and after migration.

**Comment by tinawang123** [comment]
Verified run: https://openqa.suse.de/tests/3491907

File: lib/services/registered_addons.pm (hunk 2 above)
Comment (tinawang123) [reply lib/services/registered_addons.pm]: I think we can use this way to get more exact result. If any addons status is not registered, then it can show up the addons' name. It is easy for reviewer get the failed information and report the bug.

Comment (okurz) [reply lib/services/registered_addons.pm]: I see

File: lib/services/registered_addons.pm (hunk 3 above)
Comment (okurz) [reply lib/services/registered_addons.pm]: Yes, sure. But this is not related to the goal of the single git commit

Comment (tinawang123) [reply lib/services/registered_addons.pm]: Change back as it is not related :)
